### ISSUE TITLE
Using cross-entropy loss does not penalize negative samples

### ISSUE THREAD

**Issue by xlnwel** [MishaLaskin/curl#9]
Hi,

I see that you use the cross-entropy(CE) loss for the contrastive learning. As far as I understand, this does not penalize the negative samples, as the CE loss gives zero weights to the non-diagonal entries in the [B, B] matrix. Do I make any mistake?

Best,
Sherwin

**Comment by MishaLaskin** [comment]
https://pytorch.org/docs/master/generated/torch.nn.CrossEntropyLoss.html

**Comment by MishaLaskin** [comment]
CE penalizes entries labeled with 0 (off diagonal) and encourages entries
labeled with 1 (on diagonal)

**Comment by xlnwel** [comment]
My mistake. I took `logits` as probabilities. Thanks for your help:-)
